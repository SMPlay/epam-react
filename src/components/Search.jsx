import React from "react";
import { createUseStyles } from "react-jss";
import Input from "./Input";
import Button from "./Button";
import Logo from "./Logo";

const useStyles = createUseStyles({
  buttonsContainer: {
    display: "flex",
    justifyContent: "space-between",
    alignItems: "center",
    marginTop: 20,
    "& *": {
      textTransform: "uppercase",
    },
  },
  searchBy: {
    display: "flex",
    width: 327,
    justifyContent: "space-between",
    alignItems: "center",
    marginTop: 15,
  },
  searchBySubtitle: {
    fontWeight: 700,
    color: "white",
  },
});

const Search = (props) => {
  const {
    searchBy, inputValue, onChangeInput, searchBySwitcher, submitValue,
  } = props;
  const classes = useStyles();

  return (
    <>
      <div className="row">
        <Logo />
      </div>
      <div className="row">
        <Input placeholder="Enter searching film" value={inputValue} onChange={onChangeInput} />
      </div>
      <div className="row">
        <div className={classes.buttonsContainer}>
          <div className={classes.searchBy}>
            <p className={classes.searchBySubtitle}>Search by</p>
            <Button text="title" isActive={searchBy === "title"} onClick={searchBySwitcher.bind(this, "title")} />
            <Button text="genre" isActive={searchBy === "genres"} onClick={searchBySwitcher.bind(this, "genres")} />
          </div>
          <Button onClick={submitValue} text="search" fontSize={15} width={200} background="rgb(255, 41, 41)" />
        </div>
      </div>
    </>
  );
};

export default Search;
