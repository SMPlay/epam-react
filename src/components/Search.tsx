import React from "react";
import { createUseStyles } from "react-jss";
import { Input } from "./Input";
import { Button } from "./Button";
import { Logo } from "./Logo";
import { SearchType, SortType } from "../types";

const useStyles = createUseStyles({
  buttonsContainer: {
    display: "flex",
    justifyContent: "space-between",
    alignItems: "center",
    marginTop: "20px",
    "& *": {
      textTransform: "uppercase",
    },
  },
  searchBy: {
    display: "flex",
    width: "327px",
    justifyContent: "space-between",
    alignItems: "center",
    marginTop: "15px",
  },
  searchBySubtitle: {
    fontWeight: "700px",
    color: "white",
  },
});

interface Props {
  searchBy: SearchType;
  value: string;
  onChange: (value: string) => void;
  toggleSearchBy: (type: SearchType) => void;
  submitValue: () => void;
}

export const Search: React.FC<Props> = (props) => {
  const {
    searchBy, value, onChange, toggleSearchBy, submitValue,
  } = props;
  const classes = useStyles();

  return (
    <>
      <div className="row">
        <Logo />
      </div>
      <div className="row">
        <Input
          placeholder="Enter searching film"
          value={value}
          onChange={onChange}
          dataTestAttr="search-input"
        />
      </div>
      <div className="row">
        <div className={classes.buttonsContainer}>
          <div className={classes.searchBy}>
            <p className={classes.searchBySubtitle}>Search by</p>
            <Button
              type="primary"
              active={searchBy === "title"}
              onClick={() => toggleSearchBy("title")}
              dataTestAttr="search-by-title"
            >
              &nbsp;title
            </Button>
            <Button
              type="primary"
              active={searchBy === "genres"}
              onClick={() => toggleSearchBy("genres")}
              dataTestAttr="search-by-genre"
            >
              &nbsp;genre
            </Button>
          </div>
          <Button
            type="primary"
            onClick={submitValue}
            fontSize="15px"
            width="200px"
            background="rgb(255, 41, 41)"
            dataTestAttr="search"
          >
            &nbsp;search
          </Button>
        </div>
      </div>
    </>
  );
};
