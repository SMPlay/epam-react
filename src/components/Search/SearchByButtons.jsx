import React, { useContext } from "react";
import { createUseStyles } from "react-jss";
import Button from "../Button";
import FilmsContext from "../../context/FilmsContext";

const useStyles = createUseStyles({
  searchBy: {
    display: "flex",
    width: 327,
    justifyContent: "space-between",
    alignItems: "center",
    marginTop: 15,
  },
  searchBySubtitle: {
    fontWeight: 700,
    color: "white",
  },
});

const SearchByButtons = () => {
  const { searchBy, searchBySwitch } = useContext(FilmsContext);
  const classes = useStyles();

  return (
    <div className={classes.searchBy}>
      <p className={classes.searchBySubtitle}>search by</p>
      <Button
        text="title"
        fontSize={12}
        width={100}
        isActive={searchBy === "title"}
        onClick={(e) => searchBySwitch(e, "title")}
      />
      <Button
        text="genre"
        fontSize={12}
        width={100}
        isActive={searchBy === "genre"}
        onClick={(e) => searchBySwitch(e, "genre")}
      />
    </div>
  );
};

export default SearchByButtons;
