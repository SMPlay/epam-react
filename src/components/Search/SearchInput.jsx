import React, { useContext } from "react";
import { createUseStyles } from "react-jss";
import FilmsContext from "../../context/FilmsContext";

const useStyles = createUseStyles({
  searchInput: {
    width: "100%",
    height: 35,
    padding: "0 10px",
    backgroundColor: "black",
    color: "white",
    border: "none",
    borderBottom: "1px solid rgb(255, 41, 41)",
    borderRadius: 4,
    outline: "none",
    marginTop: 10,
  },
});

const SearchInput = () => {
  const classes = useStyles();
  const { onChangeSearchInput, searchInputValue } = useContext(FilmsContext);

  return (
    <input
      className={classes.searchInput}
      placeholder="Enter searching film"
      type="text"
      value={searchInputValue}
      onChange={({ target: { value } }) => onChangeSearchInput(value)}
    />
  );
};

export default SearchInput;
