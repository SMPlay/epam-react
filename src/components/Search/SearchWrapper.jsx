import React from "react";
import Search from "containers/Search";
import Logo from "../Logo";

const SearchWrapper = () => (
  <>
    <Logo />
    <Search />
  </>
);

export default SearchWrapper;
