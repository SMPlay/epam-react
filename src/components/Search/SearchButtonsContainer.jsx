import React, { useContext } from "react";
import { createUseStyles } from "react-jss";
import Button from "../Button";
import SearchByButtons from "./SearchByButtons";
import FilmsContext from "../../context/FilmsContext";

const useStyles = createUseStyles({
  buttonsContainer: {
    display: "flex",
    justifyContent: "space-between",
    alignItems: "center",
    marginTop: 20,
    "& *": {
      textTransform: "uppercase",
    },
  },
});

const SearchButtonsContainer = () => {
  const classes = useStyles();
  const { submitValueFromInput } = useContext(FilmsContext);

  return (
    <div className={classes.buttonsContainer}>
      <SearchByButtons />
      <>
        <Button
          text="search"
          fontSize={16}
          width={200}
          background="rgb(255, 41, 41)"
          onClick={(e) => submitValueFromInput(e)}
        />
      </>
    </div>
  );
};

export default SearchButtonsContainer;
