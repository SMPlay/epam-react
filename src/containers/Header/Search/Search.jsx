import React from "react";
import { createUseStyles } from "react-jss";
import SearchButtonsContainer from "components/SearchButtonsContainer";
import SearchInput from "components/SearchInput";

const useStyles = createUseStyles({
  searchField: {
    marginTop: 50,
  },
  searchTitle: {
    fontSize: 14,
    color: "white",
    textTransform: "uppercase",
  },
});

const Search = () => {
  const classes = useStyles();

  return (
    <div className={classes.searchField}>
      <h2 className={classes.searchTitle}>find your movie</h2>
      <SearchInput />
      <SearchButtonsContainer />
    </div>
  );
};

export default Search;
