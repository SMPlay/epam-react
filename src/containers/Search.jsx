import React from "react";
import { createUseStyles } from "react-jss";
import SearchButtonsContainer from "components/Search/SearchButtonsContainer";
import SearchInput from "components/Search/SearchInput";

const useStyles = createUseStyles({
  searchField: {
    marginTop: 50,
  },
  searchTitle: {
    fontSize: 14,
    color: "white",
    textTransform: "uppercase",
  },
});

const Search = () => {
  const classes = useStyles();

  return (
    <form className={classes.searchField}>
      <h2 className={classes.searchTitle}>find your movie</h2>
      <SearchInput />
      <SearchButtonsContainer />
    </form>
  );
};

export default Search;
