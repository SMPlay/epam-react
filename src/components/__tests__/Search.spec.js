import React from "react";
import renderer from "react-test-renderer";
import { mount } from "enzyme";
import Search from "../Search";


describe("Search component", () => {
  let props; let
    component;
  beforeEach(() => {
    props = {
      searchBy: "title",
      inputValue: "",
      onChangeInput: jest.fn(),
      searchBySwitcher: jest.fn(),
      submitValue: jest.fn(),
    };

    component = mount(<Search {...props} />);
  });

  it("should render with initial state", () => {
    const tree = renderer.create(<Search {...props} />);

    expect(tree.toJSON()).toMatchSnapshot();
  });

  it("should call function on click", () => {
    component.find("Button").first().simulate("click");
    component.find("Button").at(1).simulate("click");

    expect(props.searchBySwitcher).toBeCalled();
  });

  it("should one button have active", () => {
    const firstButton = component.find("Button").first();
    const secondButton = component.find("Button").at(1);

    expect(firstButton.prop("isActive")).toBeTruthy();
    expect(secondButton.prop("isActive")).toBeFalsy();
  });
});
