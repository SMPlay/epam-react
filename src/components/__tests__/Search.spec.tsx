import React from "react";
import { BrowserRouter as Router } from "react-router-dom";
import renderer from "react-test-renderer";
import { mount } from "enzyme";
import { Search } from "../Search";
import { SearchType, SortType } from "../../types";

describe("Search component", () => {
  interface Props {
    searchBy: SearchType;
    inputValue: string;
    onChangeInput: (value: string) => void;
    toggleSearchBy: (searchBy: SearchType) => void;
    submitValue: () => void;
  }

  let props: Props;
  let component = mount<Router>(
    <Router>
      <Search {...props} />
    </Router>,
  );
  beforeEach(() => {
    props = {
      searchBy: "title",
      inputValue: "",
      onChangeInput: jest.fn(),
      toggleSearchBy: jest.fn(),
      submitValue: jest.fn(),
    };
  });
  afterEach(() => {
    component = mount<Router>(
      <Router>
        <Search {...props} />
      </Router>,
    );
  });

  it("should render with initial state", () => {
    const tree = renderer.create(
      <Router>
        <Search {...props} />
      </Router>,
    );

    expect(tree.toJSON()).toMatchSnapshot();
  });

  it("should one button have active", () => {
    const firstButton = component.find("Button").first();
    const secondButton = component.find("Button").at(1);

    expect(firstButton.prop("active")).toBeTruthy();
    expect(secondButton.prop("active")).toBeFalsy();
  });
});
