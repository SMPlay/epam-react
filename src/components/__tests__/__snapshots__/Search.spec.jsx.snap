// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search component should render with initial state 1`] = `
Array [
  <div
    className="row"
  >
    <div>
      <a
        href="/"
        onClick={[Function]}
      >
        <h1
          className="logo-0-2-4"
        >
          netflixroulette
        </h1>
      </a>
    </div>
  </div>,
  <div
    className="row"
  >
    <input
      className="searchInput-0-2-5"
      onChange={[Function]}
      placeholder="Enter searching film"
      type="text"
      value=""
    />
  </div>,
  <div
    className="row"
  >
    <div
      className="buttonsContainer-0-2-1"
    >
      <div
        className="searchBy-0-2-2"
      >
        <p
          className="searchBySubtitle-0-2-3"
        >
          Search by
        </p>
        <button
          className="btn-0-2-6 btn-d3-0-2-10 undefined active"
          onClick={[Function]}
        >
           title
        </button>
        <button
          className="btn-0-2-6 btn-d4-0-2-11 undefined "
          onClick={[Function]}
        >
           genre
        </button>
      </div>
      <button
        className="btn-0-2-6 btn-d5-0-2-12 undefined "
        onClick={[MockFunction]}
      >
         search
      </button>
    </div>
  </div>,
]
`;
